contracts/ERC5725.sol

```diff
@@ -95,16 +121,57 @@ abstract contract ERC5725 is IERC5725, ERC721 {
         return _payoutToken(tokenId);
     }
 
+    /**
+     * @dev See {IERC5725}.
+     */
+    function allowance(address owner, address spender) public view override(IERC5725) returns (uint256 result) {
+        return _allowances[owner][spender];
+    }
+
     /**
      * @dev See {IERC165-supportsInterface}.
-     * IERC5725 interfaceId = 0x7c89676d
+     * IERC5725 interfaceId = 0xf316c058
      */
     function supportsInterface(
         bytes4 interfaceId
     ) public view virtual override(ERC721, IERC165) returns (bool supported) {
         return interfaceId == type(IERC5725).interfaceId || super.supportsInterface(interfaceId);
     }
 
+    /**
+     * @dev Internal function to set a spendable allowance
+     *
+     * @param owner The owner of the token
+     * @param spender The spender who is permitted to spend the allowance
+     * @param value The allowance to be set for spender
+     *
+     */
+    function _setClaimAllowance(address owner, address spender, uint256 value) internal virtual {
+        if (spender == address(0)) {
+            revert("ERC5725: spender cannot be 0 address");
+        }
+        _allowances[owner][spender] = value;
+        emit ClaimApproval(owner, spender, value);
+    }
+
+    /**
+     * @dev Internal function to spend a set allowance
+     *
+     * @param owner The owner of the token
+     * @param spender The spender who is claiming the allowance
+     * @param value The allowance to be spent by spender
+     *
+     */
+    function _spendAllowance(address owner, address spender, uint256 value) internal virtual {
+        uint256 currentAllowance = allowance(owner, spender);
+        if (currentAllowance != type(uint256).max) {
+            if (currentAllowance < value) {
+                revert("ERC5725: insufficient allowance");
+            }
+            _allowances[owner][spender] = currentAllowance - value;
```

Comment:
With sol 0.8, we can do this safely with no under/overflow right? That's my assumption here and for the above increase/decreaseAllowance. 

The two checks here for `currentAllowance != type(uint256).max` and `currentAllowance < value` come from IERC20 [here](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/121be5dd09caa2f7ce731f0806b0e14761023bd6/contracts/token/ERC20/ERC20.sol#L361)